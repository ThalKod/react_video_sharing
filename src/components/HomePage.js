import React from "react";


export default class HomePage extends React.Component{
  render(){
    return (
        <div>
         Hello World !
        </div>
    )
  }
};
