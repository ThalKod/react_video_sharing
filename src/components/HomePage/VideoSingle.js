import React from "react";

import { request, formatSecondForRendering } from "../../utils/index";

// import videoPlaceholder from "../assets/images/video_placeholder.png";

class VideoSingle extends React.Component{

  state = {
    videoImage: null,
    // videoThumbnails: null,
  };

  componentDidMount = () => {
    const { _id } = this.props;

    request("get", `/video/cover/default/${_id}`)
        .then(res => {
          if(!res.data.error) this.setState({ videoImage: res.data.coverPhoto })
        })
        .catch(err => console.log(err));
  };

  render(){
    const { name, duration, viewCount } = this.props;
    const { videoImage } = this.state;

    return (
        <div className="col-lg-3 col-sm-6 videoitem">
          <div className="b-video">
            <div className="v-img">
              <a href="/"><img src={videoImage} alt="placeholder" /></a>
              <div className="time">{formatSecondForRendering(duration)}</div>
            </div>
            <div className="v-desc">
              <a href="/">{name}</a>
            </div>
            <div className="v-views">
              {viewCount} views.
            </div>
          </div>
        </div>
    )
  }
}

export default VideoSingle;

