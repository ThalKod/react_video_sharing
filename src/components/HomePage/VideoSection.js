import React from "react";

import VideoSingle from "components/HomePage/VideoSingle";

export default class VideoSection extends React.Component{

  renderVideoList = () => {
    const { recommendedVideos, featuredVideos, videos } = this.props;

    if(featuredVideos && featuredVideos.length > 0 ){
      return featuredVideos.map(video => {
        return <VideoSingle key={video._id} {...video} />
      })
    }

    if(recommendedVideos && recommendedVideos.length > 0){
      return recommendedVideos.map(video => {
        return <VideoSingle key={video._id} {...video} />
      })
    }

    return videos.map(video => {
      return <VideoSingle key={video._id} {...video} />
    })

  };

  onScroll = () => {
    const { getMoreVideos } = this.props;
    if (// TODO: refactor scrolling and implement a better infinite bottom scroll for this component. lets' keep this for this mvp...
        window.innerHeight + document.documentElement.scrollTop === document.documentElement.offsetHeight
    ) {
      getMoreVideos()
    }
  };

  componentDidMount = () => {
    const { scrollable } = this.props;
    if(!scrollable) return;
    window.addEventListener("scroll", this.onScroll, false);
  };

  componentWillUnmount = () => {
    const { scrollable } = this.props;
    if(!scrollable) return;
    window.removeEventListener("scroll", this.onScroll, false);
  };

  render() {
    const { type, header } = this.props;
    return (
        // TODO: make it more generics...
        <div className="content-block head-div">
          { header && <div className="cb-header">
                        <div className="row">
                          <div className="col-lg-10 col-sm-10 col-xs-8">
                            <ul className="list-inline">
                              <li><a href="/" className="color-active">{type}</a></li>
                            </ul>
                          </div>
                        </div>
                      </div> }
          <div className="cb-content videolist">
            <div className="row">
              {this.renderVideoList()}
            </div>
          </div>
        </div>
    )
  }
}

