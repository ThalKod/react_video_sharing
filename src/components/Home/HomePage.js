import React from "react";
import { connect } from "react-redux";

import VideoSection from "components/Home/VideoSection";
import LoadingSpinner from "components/Common/LoadingSpinner";
import { startGetRecommendedVideo, startGetVideos } from "actions";

export class HomePage extends React.Component{

  state = {
    loading: true
  };

  componentDidMount = () => {
    const { getRecommendedVideo, getVideos, offset } = this.props;

    const getRecommendedVideoPromise = getRecommendedVideo();
    const getVideosPromise = getVideos({ offset });

    Promise.all([ getRecommendedVideoPromise , getVideosPromise ])
        .then(res => {
          if(res[0].error === false && res[1].error === false) this.setState({ loading: false });
        })
        .catch(err => console.log(err));
  };

  render(){
    const { loading } = this.state;
    const { recommendedVideos, featuredVideos, getVideos, offset } = this.props;

    // temp loading..
    if(loading) return <LoadingSpinner/>;

    return (
        <div className="content-wrapper">
          <div className="container">
            <div className="row">
              <div className="col-lg-12">
               <VideoSection recommendedVideos={recommendedVideos} type="Recommended" header />
               <VideoSection
                   featuredVideos={featuredVideos}
                   scrollable type="Featured"
                   header
                   getMoreVideos={() => getVideos({ offset })}
               />
              </div>
            </div>
          </div>
        </div>
    )
  }
}

const mapStateToProps = (state) => ({
  recommendedVideos: state.video.recommended,
  featuredVideos: state.video.featured.videos,
  offset: state.video.featured.offset,
});

const mapDispatchToProps = (dispatch) => ({
    getRecommendedVideo: () => dispatch(startGetRecommendedVideo()),
    getVideos: (options) => dispatch(startGetVideos(options)),
});

export default connect(mapStateToProps,mapDispatchToProps)(HomePage);
