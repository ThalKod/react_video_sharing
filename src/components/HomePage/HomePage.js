import React from "react";
import { connect } from "react-redux";

import VideoSection from "./VideoSection";
import { startGetRecommendedVideo, startGetVideos } from "../../actions/index";

export class HomePage extends React.Component{

  state = {
    loading: true
  };

  componentDidMount = () => {
    const { getRecommendedVideo, getVideos } = this.props;

    const getRecommendedVideoPromise = getRecommendedVideo();
    const getVideosPromise = getVideos({});

    Promise.all([ getRecommendedVideoPromise , getVideosPromise ])
        .then(res => {
          if(res[0].error === false && res[1].error === false) this.setState({ loading: false });
        })
        .catch(err => console.log(err));
  };

  render(){
    const { loading } = this.state;

    // temp loading..
    if(loading) return <div>Loading...</div>;

    return (
        <div className="content-wrapper">
          <div className="container">
            <div className="row">
              <div className="col-lg-12">
               <VideoSection type="Recommended" />
               <VideoSection scrollable type="Featured"/>
              </div>
            </div>
          </div>
        </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
    getRecommendedVideo: () => dispatch(startGetRecommendedVideo()),
    getVideos: (options) => dispatch(startGetVideos(options)),
});

export default connect(null,mapDispatchToProps)(HomePage);
