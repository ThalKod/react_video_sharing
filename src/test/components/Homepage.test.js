import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import HomePage from "components/HomePage/HomePage";

describe("HomePage component", () => {
  it("should render the Homepage correctly", () => {
    const wrapper = shallow(<HomePage />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
