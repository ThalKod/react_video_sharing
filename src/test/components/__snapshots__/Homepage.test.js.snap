// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component should render the Homepage correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
